daemon/logger/awslogs/cloudwatchlogs.go

```diff
@@ -184,6 +185,38 @@ func (l *logStream) Close() error {
 
 // create creates a log stream for the instance of the awslogs logging driver
 func (l *logStream) create() error {
+
+	// Create log group first
+	{
+		input := &cloudwatchlogs.CreateLogGroupInput{
+			LogGroupName: aws.String(l.logGroupName),
+		}
+
+		_, err := l.client.CreateLogGroup(input)
```

Comment:
I'm worried about the latency cost of an additional round-trip for every container start.  I think better behavior would be attempting `CreateLogStream` first, and falling back to `CreateLogGroup` only when the log group does not already exist.